OpenMetrics parser crashes for single-byte `#` input

The following reproduction:

```go
package main

import (
	"github.com/prometheus/prometheus/pkg/textparse"
)

func main() {
	p := textparse.New([]byte("#"), "application/openmetrics-text")
	var err error
	for {
		_, err = p.Next()
		if err != nil {
			break
		}
	}
}
```

...leads to:

```
$ go run main.go
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/prometheus/prometheus/pkg/textparse.(*openMetricsLexer).cur(...)
	/home/julius/gosrc/src/github.com/prometheus/prometheus/pkg/textparse/openmetricsparse.go:47
github.com/prometheus/prometheus/pkg/textparse.(*OpenMetricsParser).Next(0xc0000a2000, 0x1, 0x1, 0x4f819b)
	/home/julius/gosrc/src/github.com/prometheus/prometheus/pkg/textparse/openmetricsparse.go:296 +0x175f
main.main()
	/home/julius/tmp/fuzz/main.go:11 +0x93
exit status 2
```

@yevgenypats You will be interested to know that this was discovered as the first fuzz-testing result by fuzzit (https://app.fuzzit.dev/orgs/prometheus/targets/promql-parse-open-metric/m6EqtgmNqwfLySMvwJNX) :)

--- a/pkg/textparse/openmetricsparse.go
+++ b/pkg/textparse/openmetricsparse.go
@@ -44,7 +44,10 @@ func (l *openMetricsLexer) buf() []byte {
 }
 
 func (l *openMetricsLexer) cur() byte {
-	return l.b[l.i]
+	if l.i < len(l.b) {
+		return l.b[l.i]
+	}
+	return byte(' ')
 }
 
 // next advances the openMetricsLexer to the next character.

--- a/pkg/textparse/openmetricsparse_test.go
+++ b/pkg/textparse/openmetricsparse_test.go
@@ -420,6 +420,14 @@ func TestOMNullByteHandling(t *testing.T) {
 			input: "a\x00{b=\"ddd\"} 1",
 			err:   "expected value after metric, got \"MNAME\"",
 		},
+		{
+			input: "#",
+			err:   "\"INVALID\" \" \" is not a valid start token",
+		},
+		{
+			input: "# H",
+			err:   "\"INVALID\" \" \" is not a valid start token",
+		},
 	}
 
 	for i, c := range cases {

